### ISSUE TITLE
RFE: Allow triggering actions (like sending email) based on journal messages

### ISSUE THREAD

**Issue by rmunn** [systemd/systemd#5431]
### Submission type

  - [ ] Bug report
  - [X] Request for enhancement (RFE)

### systemd version the issue has been seen with

Is this relevant for an RFE? In case it is, I'm using Ubuntu Xenial, whose current `systemd` package is version `229-4ubuntu16`.

### Used distribution

Ubuntu 16.04.2 LTS (Xenial)

### Details of requested enhancement

I'm surprised this hasn't (AFAICT) been requested before (I searched closed and new issues for words like "trigger", "email", "journal", etc.). I'd like to see a way to trigger certain actions when messages matching certain criteria are written to the journal. For example, sending an email when a message of severity Critical or higher comes in.

Currently this need is partially filled by third-party projects like https://github.com/jjk-jacky/journal-triggerd, but it would be nice to have an "official" way of doing it.

The details of how the triggers would be defined, what actions could be taken on them, and so on, would have to be hammered out, so I'm only talking in the vaguest of terms about what they could be. But some examples might include:

1. When a log message arrives with syslog target name `MyApp` and text containing `Exception`, email the message body to `crashreports-myapp@example.com`.
1. When a log message arrives from any source with priority `Critical`, send an HTTP POST to `https://example.com/api/urgent-logs` with that log message in JSON format.
1. When more than N log messages arrive from source X within M milliseconds, pause logging from source X for S seconds and email `sysadmin-for-server-x@example.com` with message body "Server X is spamming log messages again."

**Comment by poettering** [comment]
Yeah, activation by journal message has been on the TODO list for a while.
